functions-and-scope.js

```diff
@@ -1,84 +1,107 @@
-// * Hoe kan ik iedere waarde van de array langsgaan?
-// * Op welke conditie moet ik checken?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
-// Log het antwoord in de terminal.
-
-// ---- Verwachte uitkomst: 9
-
-
-/* 3b: Omschrijven tot een herbruikbare functie */
-// Schrijf een functie genaamd highestGrade, die een array van cijfers verwacht (zoals grades) en het hoogste cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 3a.
-// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
-// Log het antwoord in de terminal.
-
+// typeOfEmail("n.eeken@novi-education.nl") geeft "Student"
+// typeOfEmail("t.mellink@novi.nl") geeft geeft "Medewerker"
+// typeOfEmail("novi.nlaapjesk@outlook.com") geeft geeft "Extern" <-- deze moet het ook doen!
+// typeOfEmail("a.wiersma@outlook.com") geeft "Extern"
+
+// Declaring function to check the type of email address
+function typeOfEmail (mail) {
+    // First split address at @, then check content after @
+    if (mail.split("@")[1] === "novi-education.nl") {
+        return "Student"
+    }
+    else if (mail.split("@")[1] === "novi.nl") {
+        return "Medewerker"
+    }
+    else {
+        return "Extern"
+    }
+}
+
+// Calling function and test if it works for different addresses
+console.log(typeOfEmail(myMail))
+console.log(typeOfEmail("n.eeken@novi-education.nl"))
+console.log(typeOfEmail("t.mellink@novi.nl"))
+console.log(typeOfEmail("a.wiersma@outlook.com"))
+
+/* Opdracht  3 */
+// Schrijf een functie genaamd checkEmailValidity, die een emailadres verwacht en checkt of het emailadres valide is. De functie returned true of false, afhankelijk van de uitkomst.
+// Een emailadres is valide wanneer:
+// * Er een @ in voorkomt
+// * Er géén , in voorkomt
+// * Er géén . in voorkomt als allerlaatste karakter (dus hotmail.com is valide, net als outlook.nl, maar outlooknl. niet)
 // ---- Verwachte uitkomsten:
-// highestGrade(grades) geeft 9
-// highestGrade([6, 4, 5]) geeft 6
-// highestGrade([8, 9, 4, 6, 10]) geeft 10
+// checkEmailValidity("n.eeken@novi.nl") geeft true - want @ en punt op de juiste plek
+// checkEmailValidity("tessmellink@novi.nl") geeft true - want @ en punt op de juiste plek
+// checkEmailValidity("n.eekenanovi.nl") geeft false - want geen @
+// checkEmailValidity("n.eeken@novinl.") geeft false - want de punt mag niet als laatst
+// checkEmailValidity("tessmellink@novi,nl") geeft false - want er staat een komma in
+
+// Declaring function to check if email address has a valid format
+function checkEmailValidity(mail) {
+    let isValid = true;
+    if ((mail.indexOf("@") === -1) ||
+        mail.includes(",") ||
+        (mail.lastIndexOf(".") === (mail.length-1))) {
+        isValid = false
+    }
+    return isValid;
```

Comment:
Je zou dit ook in één keer terug kunnen geven, maar dan natuurlijk wel de inverse:
`return mail.includes("@") && !mail.includes(",") && mail[mail.length-1] === ".";`